### PR TITLE
Core/Achievements: Fix CriteriaTypes related to GameEvents

### PR DESCRIPTION
**Pull request by meji46**
**Changes proposed:**

-  Added missing cases in CriteriaHandler::RequirementsSatisfied for CriteriaType::PlayerTriggerGameEvent and CriteriaType::AnyoneTriggerGameEventScenario (all GameEvent criteria without additional requirements were adding up to 1 in each login)
-  Criteria GameEvents added to _eventStore

**Issues addressed:**
None

**Tests performed:**
Builds, tested in-game

**Known issues and TODO list:** (add/remove lines as needed)
None

### CODE REVIEW

File: src/server/game/Globals/ObjectMgr.cpp [hunk 1]
```diff
@@ -16,6 +16,7 @@
  */
 
 #include "ObjectMgr.h"
+#include "AchievementMgr.h"
```
Comment (Shauren) [inline src/server/game/Globals/ObjectMgr.cpp:19]: shouldnt this be `CriteriaHandler.h` (thats where `sCriteriaMgr` lives)

Comment (meji46) [reply src/server/game/Globals/ObjectMgr.cpp]: Yep, you're right. Fixed!

**Comment by Shauren** [comment]
There are conflicts here caused by merging 10b06ff50f77cc0516c5f17adffef8eb75f1d555

**Comment by meji46** [comment]
Fixed

**Pull request merged**